src/transport/SecurePairingSession.cpp

```diff
@@ -0,0 +1,310 @@
+        SuccessOrExit(err);
+    }
+
+    return err;
+
+exit:
+
+    mNextExpectedMsg = Spake2pMsgType::kSpake2pMsgTypeMax;
+    return err;
+}
+
+CHIP_ERROR SecurePairingSession::HandleCompute_pB_cB(const MessageHeader & header, System::PacketBuffer * msg)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    uint8_t verifier[kMAX_Hash_Length];
+    size_t verifier_len = kMAX_Hash_Length;
+
+    const uint8_t * buf = msg->Start();
+    size_t buf_len      = msg->TotalLength();
+
+    VerifyOrExit(buf != nullptr, err = CHIP_ERROR_MESSAGE_INCOMPLETE);
+    VerifyOrExit(buf_len == kMAX_Point_Length + kMAX_Hash_Length, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+
+    err = mSpake2p.ComputeRoundTwo(buf, kMAX_Point_Length, verifier, &verifier_len);
+    SuccessOrExit(err);
+
+    {
+        // Call delegate to send the Compute_cA to peer
+        System::PacketBuffer * resp = System::PacketBuffer::NewWithAvailableSize(verifier_len);
+        VerifyOrExit(resp != nullptr, err = CHIP_SYSTEM_ERROR_NO_MEMORY);
+
+        memcpy(resp->Start(), verifier, verifier_len);
+        resp->SetDataLength(verifier_len);
+
+        err = mDelegate->OnNewMessageForPeer(Spake2pMsgType::kSpake2pCompute_cA, resp);
+        SuccessOrExit(err);
+    }
+
+    {
+        const uint8_t * hash = &buf[kMAX_Point_Length];
+        err                  = mSpake2p.KeyConfirm(hash, kMAX_Hash_Length);
+        SuccessOrExit(err);
+
+        err = mSpake2p.GetKeys(mKe, &mKeLen);
+        SuccessOrExit(err);
+    }
+
+    // Call delegate to indicate pairing completion
+    mDelegate->OnPairingComplete();
+    mPairingComplete = true;
+
+exit:
+
+    mNextExpectedMsg = Spake2pMsgType::kSpake2pMsgTypeMax;
+    return err;
+}
+
+CHIP_ERROR SecurePairingSession::HandleCompute_cA(const MessageHeader & header, System::PacketBuffer * msg)
+{
+    CHIP_ERROR err       = CHIP_NO_ERROR;
+    const uint8_t * hash = msg->Start();
+
+    VerifyOrExit(hash != nullptr, err = CHIP_ERROR_MESSAGE_INCOMPLETE);
+    VerifyOrExit(msg->TotalLength() == kMAX_Hash_Length, err = CHIP_ERROR_INVALID_MESSAGE_LENGTH);
+
+    err = mSpake2p.KeyConfirm(hash, kMAX_Hash_Length);
+    SuccessOrExit(err);
+
+    err = mSpake2p.GetKeys(mKe, &mKeLen);
+    SuccessOrExit(err);
+
+    // Call delegate to indicate pairing completion
+    mDelegate->OnPairingComplete();
+    mPairingComplete = true;
+
+exit:
+
+    mNextExpectedMsg = Spake2pMsgType::kSpake2pMsgTypeMax;
+    return err;
+}
+
+CHIP_ERROR SecurePairingSession::HandlePeerMessage(const MessageHeader & header, System::PacketBuffer * msg)
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+    VerifyOrExit(msg != nullptr, err = CHIP_ERROR_INVALID_ARGUMENT);
+    VerifyOrExit(header.GetMessageType() == (uint8_t) mNextExpectedMsg, err = CHIP_ERROR_INVALID_MESSAGE_TYPE);
+
+    // The protocol for handling pA, pB, cB and cA is defined in SPAKE2 Plus specifications
+    // (https://www.ietf.org/id/draft-bar-cfrg-spake2plus-01.html)
```

Comment:
Also, I would suggest moving this specification reference up to the `@file` comment at the top of this file.